### PR TITLE
[5.2] Validate app name command

### PR DESCRIPTION
**Pull request by ahmedash95**

### CODE REVIEW

File: src/Illuminate/Foundation/Console/AppNameCommand.php [hunk 1]
```diff
@@ -67,6 +67,10 @@ public function __construct(Composer $composer, Filesystem $files)
      */
     public function fire()
     {
+        if (! preg_match('/^[A-Za-z][A-Za-z0-9]*(?:_[A-Za-z0-9]+)*$/', $this->argument('name'))) {
+            return $this->error('Invalid App Name');
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Console/AppNameCommand.php]: Might be better to just throw an exception here.

Comment (ahmedash95) [inline src/Illuminate/Foundation/Console/AppNameCommand.php]: it throws exception now , but i don't know what's the problem for using AppNameCommandException with StyleCI

File: src/Illuminate/Foundation/Console/AppNameCommandException.php [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Illuminate\Foundation\Console;
+
+use Exception;
+
+class AppNameCommandException extends Exception
+{
+    public function __construct($message)
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Console/AppNameCommandException.php]: Don't do that please.

File: src/Illuminate/Foundation/Console/AppNameCommand.php (hunk 1 above)
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Console/AppNameCommand.php]: That's because you don't need to import it.

**Comment by GrahamCampbell** [comment]
I don't think we need another exception class here. Just throw RuntimeException or something.

**Comment by ahmedash95** [comment]
@GrahamCampbell what about \Exception class only ?

File: src/Illuminate/Foundation/Console/AppNameCommandException.php (hunk 2 above)
Comment (ahmedash95) [inline src/Illuminate/Foundation/Console/AppNameCommandException.php]: i just have a question here .. why not ? a lot of laravel's exceptions has it's own exception class, i'm trying to understand why you are doing that but your previous comment make me confuse when to make it and when not ?

**Comment by GrahamCampbell** [comment]
Just leave it with something basic, and Taylor will just change it to whatever he wants. Please squash to one commit too.

**Comment by crynobone** [comment]
> Please squash to one commit too.

:trollface: https://github.com/blog/2141-squash-your-commits

File: src/Illuminate/Foundation/Console/AppNameCommandException.php (hunk 2 above)
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Console/AppNameCommandException.php]: I'm referring to the pointless constructor method.

**Pull request closed without merging**